include/sundials/sundials_stepper.h

```diff
@@ -0,0 +1,113 @@
+/* -----------------------------------------------------------------
+ * SUNDIALS Copyright Start
+ * Copyright (c) 2002-2024, Lawrence Livermore National Security
+ * and Southern Methodist University.
+ * All rights reserved.
+ *
+ * See the top-level LICENSE and NOTICE files for details.
+ *
+ * SPDX-License-Identifier: BSD-3-Clause
+ * SUNDIALS Copyright End
+ * -----------------------------------------------------------------*/
+
+#ifndef _SUNDIALS_STEPPER_H
+#define _SUNDIALS_STEPPER_H
+
+#include <sundials/sundials_core.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+typedef int (*SUNJacFn)(sunrealtype t, N_Vector y, N_Vector fy, SUNMatrix Jac,
+                        void* user_data, N_Vector tmp1, N_Vector tmp2,
+                        N_Vector tmp3);
+
+typedef int (*SUNJacTimesFn)(N_Vector v, N_Vector Jv, sunrealtype t, N_Vector y,
+                             N_Vector fy, void* user_data, N_Vector tmp);
```

Comment:
In case we need similar functions for DAEs in the future
```suggestion
typedef int (*SUNODEJacFn)(sunrealtype t, N_Vector y, N_Vector fy, SUNMatrix Jac,
                           void* user_data, N_Vector tmp1, N_Vector tmp2,
                           N_Vector tmp3);

typedef int (*SUNODEJacTimesFn)(N_Vector v, N_Vector Jv, sunrealtype t, N_Vector y,
                                N_Vector fy, void* user_data, N_Vector tmp);
```

Reply:
We could do this, or we could just name the DAE ones `SUNResJacFn` etc if we add support for them later.